lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -705,17 +738,54 @@ sub get_dataset_allele_response {
       $externalURL = "http://grch37.ensembl.org";
     }
 
-    if($sv == 1 && defined $vf) {
-      $externalURL .= "/Homo_sapiens/StructuralVariation/Explore?sv=" . $vf->variation_name();
-    }
-    elsif($sv == 0 && defined $vf) {
-      $externalURL .= "/Homo_sapiens/Variation/Explore?v=" . $vf->name();
+    my $count = 0;
+
+    if($vf) {
+      my $url = '';
+      my $var_name;
+      my $delimiter;
+      foreach my $variant (@{$vf}) {
+        if($sv == 1) {
+          $var_name = $variant->variation_name();
+          $delimiter = "StructuralVariation/Explore?sv=";
+        }
+        else {
+          $var_name = $variant->name();
+          $delimiter = "Variation/Explore?v=";
+        }
+
+        # Checks if dataset_id is one of the datasets where variant is found
+        # If it's not found then dataset response won't include this variant
+        my %var_dataset = %{$variant_dt};
+        my $datasets = $var_dataset{$var_name};
+        my $contains = contains_value($dataset_id, $datasets);
+        if($contains) {
+          $count += 1;
+          $url .= ', ' . $externalURL . "/Homo_sapiens/" . $delimiter . $var_name;
+        }
+      }
+      $url =~ s/, //;
```

Comment:
why are we adding ', ' to the $url just to remove it later?
am I missing something?

Reply:
oh I see now...
@nerdstrike is correct, building a `@url` would be a better way of doing this, and would remove the need for `$count` (that's the size of the array now) and just join it in the end to build the mega url string.

Reply:
GA4GH doesn't expect multiple URLs, so I didn't know where to put the different URLs. Merging them seemed ok but I'm also unsure what's the best approach.

Reply:
Tell them their spec is broken for real data. You shouldn't violate the schema as it stands, so the workaround is the only option at this time.